Tests/Tests/RouterSpec.swift

```diff
@@ -217,6 +260,45 @@ class RouterSpec: QuickSpec {
                 }
             }
 
+            context("when calling isDisplayingTooltip") {
+
+                let tooltipTargetView = UIView(frame: CGRect(x: 100, y: 100, width: 10, height: 10))
+                let tooltip = TestHelpers.generateTooltip(id: "test")
+                let imageBlob: Data! = UIImage(named: "test-image", in: .unitTests, with: nil)?.pngData()
+
+                beforeEach {
+                    tooltipTargetView.accessibilityIdentifier = TooltipViewIdentifierMock
+                    window.addSubview(tooltipTargetView)
+                }
+
+                afterEach {
+                    tooltipTargetView.removeFromSuperview()
+                    window.findTooltipView()?.removeFromSuperview()
+                }
+
+                it("will return true when tooltip displayed") {
+                    router.displayTooltip(tooltip, targetView: tooltipTargetView,
+                                          identifier: TooltipViewIdentifierMock,
+                                          imageBlob: imageBlob,
+                                          becameVisibleHandler: { _ in },
+                                          confirmation: true,
+                                          completion: { _ in })
+                    expect(window.findTooltipView()).toEventuallyNot(beNil())
+                    expect(router.isDisplayingTooltip(with: TooltipViewIdentifierMock)).to(beTrue())
+                }
+
+                it("will return false when tooltip is not displayed") {
+                    expect(router.isDisplayingTooltip(with: "test")).to(beFalse())
+                }
+
+                it("will return a value when calling the function from main thread") {
+                    let q = DispatchQueue(label: "test")
+                    q.async {
+                        expect(router.isDisplayingTooltip(with: "test")).to(beFalse())
```

Comment:
This will not be executed because test will not wait for this thread. You need to use `waitUntil` block and call `done()` after `expect`

Reply:
Updated here [f60c555](https://github.com/rakutentech/ios-inappmessaging/pull/275/commits/f60c555517519f44c4bd47352520a9940b80058f).